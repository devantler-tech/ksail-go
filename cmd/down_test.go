package cmd_test

import (
	"testing"

	"github.com/devantler-tech/ksail-go/cmd"
	"github.com/devantler-tech/ksail-go/cmd/internal/testutils"
)

func TestNewDownCmd(t *testing.T) {
	t.Parallel()

	testutils.TestSimpleCommandCreation(t, testutils.SimpleCommandTestData{
		CommandName:   "down",
		NewCommand:    cmd.NewDownCmd,
		ExpectedUse:   "down",
		ExpectedShort: "Destroy a cluster",
	})
}

func TestDownCmd_Execute(t *testing.T) {
	t.Parallel()

	testutils.TestSimpleCommandExecution(t, testutils.SimpleCommandTestData{
		CommandName: "down",
		NewCommand:  cmd.NewDownCmd,
	})
}

func TestDownCmd_Help(t *testing.T) {
	t.Parallel()

	testutils.TestSimpleCommandHelp(t, testutils.SimpleCommandTestData{
		CommandName: "down",
		NewCommand:  cmd.NewDownCmd,
	})
}
