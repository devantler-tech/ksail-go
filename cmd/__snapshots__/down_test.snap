
[TestDownCmdExecute - 1]
✔ cluster destroyed successfully
► Distribution: Kind
► Context: kind-ksail-default

---

[TestDownCmdHelp - 1]
Destroy a cluster.

Usage:
  down [flags]

Flags:
  -c, --context string              Kubernetes context of cluster to destroy (default "kind-ksail-default")
  -d, --distribution Distribution   Kubernetes distribution to use (default Kind)
  -h, --help                        help for down

---
